src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.cs

```diff
@@ -379,9 +379,8 @@ bool isEnabledWithAnalyzerConfigOptions(DiagnosticDescriptor descriptor)
                 // Compilation wide user settings (diagnosticOptions) from ruleset/nowarn/warnaserror overrides the analyzer author and global editorconfig settings.
                 // Note that "/warnaserror-:DiagnosticId" adds a diagnostic option with value 'ReportDiagnostic.Default',
                 // which should not alter 'isSuppressed'.
-                if ((diagnosticOptions.TryGetValue(diag.Id, out var severity) ||
-                    options.SyntaxTreeOptionsProvider is object && options.SyntaxTreeOptionsProvider.TryGetGlobalDiagnosticValue(diag.Id, cancellationToken, out severity)) &&
-                    severity != ReportDiagnostic.Default)
+                if (diagnosticOptions.TryGetValue(diag.Id, out var severity) && severity != ReportDiagnostic.Default ||
```

Comment:
I'd prefer if we added parenthesis for clarity here. I don't want to remember whether && or || binds tighter in order to read this code.

Reply:
Thanks, added